core/src/main/java/org/infinispan/cache/impl/CacheImpl.java

```diff
@@ -1863,8 +1863,8 @@ public V putIfAbsent(K key, V value, Metadata metadata) {
       return putAsync(key, value, metadata, EnumUtil.EMPTY_BIT_SET, getInvocationContextWithImplicitTransaction(false, 1));
    }
 
-   private Transaction suspendOngoingTransactionIfExists() {
-      final Transaction tx = getOngoingTransaction();
+   private Transaction suspendOngoingTransactionIfExists(boolean skipBatching) {
```

Comment:
I don't think this is (or should be) ever called with `skipBatching == false`, so I suggest removing the parameter.

I would also suggest reversing the parameter for `getOngoingTransaction`, into `includeBatchTx` (or even `includeSuspendedBatchTx`).